cmd/main.go

```diff
@@ -2,45 +2,122 @@ package main
 
 import (
 	"context"
+	"errors"
 	"flag"
-	"log"
+	"fmt"
+	"net/http"
 	"os"
 
 	"github.com/joho/godotenv"
 
 	"github.com/benchttp/server"
 	"github.com/benchttp/server/firestore"
+	"github.com/benchttp/server/shutdown"
 )
 
-const defaultPort = "9998"
+const (
+	defaultConfigPath = ".env"
+	defaultPort       = "9998"
+)
 
 func main() {
-	port := flag.String("port", defaultPort, "Address for the server to listen on.")
-	flag.Parse()
-	addr := ":" + *port
+	ctx, cancel := context.WithCancel(context.Background())
+	go shutdown.ListenInterrupt(cancel)
 
-	err := godotenv.Load(".env")
+	shutdownHandle, err := run(ctx, cancel)
 	if err != nil {
-		log.Fatalf("Error loading .env file")
+		shutdownHandle.Call() // nolint
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
 	}
 
-	projectID := os.Getenv("GOOGLE_PROJECT_ID")
-	if projectID == "" {
-		log.Fatalf("GOOGLE_PROJECT_ID variable is not defined")
+	// Wait for interrupt.
+	<-ctx.Done()
+
+	if err := shutdownHandle.Call(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
 	}
+}
+
+// run executes the main program. It starts a server.Server inside
+// a goroutine and returns a configured shutdown.Handle for gracefully
+// shutting down the program.
+//
+// If the bootstrap of the server fails, run returns a non-nil error.
+// Otherwise the caller must use the shutdown.Handle to stop the server.
+func run(ctx context.Context, cancel context.CancelFunc) (*shutdown.Handle, error) {
+	configPath := flag.String("config", defaultConfigPath, "Path to the configuration file (for example --config .env)")
```

Comment:
[nitpick] It's more an environment file than a server config to me
```suggestion
	envFile := flag.String("envFile", defaultEnvFile, "Path to environment file (for example -envFile .env)")
```